### ISSUE TITLE
Distro not supported: "Container Linux by CoreOS" 1235.6.0"

### ISSUE THREAD

**Issue by fcatae** [Azure/azure-docker-extension#116]
I can't install Docker Extension in CoreOS Azure VM. It fails with [Distro not supported: "Container Linux by CoreOS" 1235.6.0"]

I am not sure whether it is expected. However, this is the default CoreOS distribution in Azure Marketplace (https://portal.azure.com/?l=en.en-us#create/CoreOS.CoreOSStable-ARM)

/etc/lsb-release:
DISTRIB_ID="Container Linux by CoreOS"
DISTRIB_RELEASE=1235.6.0
DISTRIB_CODENAME="Ladybug"
DISTRIB_DESCRIPTION="Container Linux by CoreOS 1235.6.0 (Ladybug)"

Log files:
[azure-docker-extension-enable.txt](https://github.com/Azure/azure-docker-extension/files/711831/azure-docker-extension-enable.txt)

[waagent.txt](https://github.com/Azure/azure-docker-extension/files/711833/waagent.txt)

**Comment by colemickens** [comment]
Looks like the distro detection needs to be updated: https://github.com/Azure/azure-docker-extension/blob/0539b864cc6fd64ba1236a60443c006019eb7f10/pkg/driver/driver.go#L25

**Comment by szwed** [comment]
Same issue here. A temporary workaround for me was:

```
-              "version": "latest"
+              "version": "899.17.0"
```

**Comment by boumenot** [comment]
Here's the complete error message to help future searchers.

The log file is located in /var/log/azure/Microsoft.Azure.Extensions.DockerExtension/**<version>**/docker-extension.log.

```text
[DockerExtension] 2017/01/19 21:23:02 ----------------------------------------
[DockerExtension] 2017/01/19 21:23:02 Extension handler launch args: "/var/lib/waagent/Microsoft.Azure.Extensions.DockerExtension-1.1.1606092330/bin/docker-extension install"
[DockerExtension] 2017/01/19 21:23:02 seqnum: 0
[DockerExtension] 2017/01/19 21:23:02 Status 'transitioning' not reported for operation 'Install Docker' (by design)
[DockerExtension] 2017/01/19 21:23:02 distro info: "Container Linux by CoreOS" 1235.6.0
[DockerExtension] 2017/01/19 21:23:02 ERROR: Distro not supported: "Container Linux by CoreOS" 1235.6.0
[DockerExtension] 2017/01/19 21:23:02 Status 'error' not reported for operation 'Install Docker' (by design)
[DockerExtension] 2017/01/19 21:23:02 Cleaned up .seqnum file.
[DockerExtension] 2017/01/19 21:23:02 Exiting with code 1.
```

**Comment by szwed** [comment]
Just got an email from Microsoft, they are already working on that.

**Comment by boumenot** [comment]
I published 1.2.2 to our canary region.  If the telemetry looks good, I will publish it to West Central US.  If the telemetry continues to look good I'll publish it to all public regions.  There is a minimum delay of 24 hours between publish steps.

Once that has completed I will publish it to the other sovereign clouds.

I'll update this thread with any new status.

**Comment by boumenot** [comment]
Deployment of 1.2.2 has completed for the public cloud as of 2017-01-23T23:49:48Z, or roughly 1.25 hours ago.  The extension should be publicly visible in all regions within the next hour.  

I spot checked availability, and it was available in westus, westcentralus, southcentralus, but not yet in eastus and eastus2.

```
>azure vm extension-image list southcentralus Microsoft.Azure.Extensions DockerExtension
info:    Executing command vm extension-image list
+ Getting virtual machine extension image verions (Publisher: "Microsoft.Azure.Extensions" Type:"DockerExtension" Location:"southcentralus")
data:    Publisher                   Type             Version         Location
data:    --------------------------  ---------------  --------------  --------------
data:    Microsoft.Azure.Extensions  DockerExtension  1.0.1512030601  southcentralus
data:    Microsoft.Azure.Extensions  DockerExtension  1.1.1512090359  southcentralus
data:    Microsoft.Azure.Extensions  DockerExtension  1.1.1512180541  southcentralus
data:    Microsoft.Azure.Extensions  DockerExtension  1.1.1601070410  southcentralus
data:    Microsoft.Azure.Extensions  DockerExtension  1.1.1601140348  southcentralus
data:    Microsoft.Azure.Extensions  DockerExtension  1.1.1602270800  southcentralus
data:    Microsoft.Azure.Extensions  DockerExtension  1.1.1604142300  southcentralus
data:    Microsoft.Azure.Extensions  DockerExtension  1.1.1606092330  southcentralus
data:    Microsoft.Azure.Extensions  DockerExtension  1.2.0           southcentralus
data:    Microsoft.Azure.Extensions  DockerExtension  1.2.2           southcentralus
info:    vm extension-image list command OK
```
